solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerManager.java

```diff
@@ -0,0 +1,128 @@
+
+  private final Map<CircuitBreakerType, CircuitBreaker> circuitBreakerMap = new HashMap<>();
+
+  // Allows replacing of existing circuit breaker
+  public void registerCircuitBreaker(CircuitBreakerType circuitBreakerType, CircuitBreaker circuitBreaker) {
+    circuitBreakerMap.put(circuitBreakerType, circuitBreaker);
+  }
+
+  public CircuitBreaker getCircuitBreaker(CircuitBreakerType circuitBreakerType) {
+    assert circuitBreakerType != null;
+
+    return circuitBreakerMap.get(circuitBreakerType);
+  }
+
+  /**
+   * Check if any circuit breaker has triggered.
+   * @return CircuitBreakers which have triggered, null otherwise
+   */
+  public Map<CircuitBreakerType, CircuitBreaker> checkAllCircuitBreakersAndReturnTrippedBreakers() {
+    Map<CircuitBreakerType, CircuitBreaker> triggeredCircuitBreakers = null;
+
+    for (Map.Entry<CircuitBreakerType, CircuitBreaker> entry : circuitBreakerMap.entrySet()) {
+      CircuitBreaker circuitBreaker = entry.getValue();
+
+      if (circuitBreaker.isCircuitBreakerEnabled() &&
+          circuitBreaker.isCircuitBreakerGauntletTripped()) {
+        if (triggeredCircuitBreakers == null) {
+          triggeredCircuitBreakers = new HashMap<>();
+        }
+
+        triggeredCircuitBreakers.put(entry.getKey(), circuitBreaker);
+      }
+    }
+
+    return triggeredCircuitBreakers;
+  }
+
+  /**
+   * Returns true if *any* circuit breaker has triggered, false if none have triggered
+   *
+   * NOTE: This method short circuits the checking of circuit breakers -- the method will
+   * return as soon as it finds a circuit breaker that is enabled and has triggered
+   */
+  public boolean checkAllCircuitBreakers() {
+    for (Map.Entry<CircuitBreakerType, CircuitBreaker> entry : circuitBreakerMap.entrySet()) {
+      CircuitBreaker circuitBreaker = entry.getValue();
+
+      if (circuitBreaker.isCircuitBreakerEnabled() &&
+          circuitBreaker.isCircuitBreakerGauntletTripped()) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Construct the final error message to be printed when circuit breakers trip
+   * @param circuitBreakerMap Input list for circuit breakers
+   * @return Constructed error message
+   */
+  public static String constructFinalErrorMessageString(Map<CircuitBreakerType, CircuitBreaker> circuitBreakerMap) {
+    assert circuitBreakerMap != null;
+
+    StringBuilder sb = new StringBuilder();
+
+    for (CircuitBreakerType circuitBreakerType : circuitBreakerMap.keySet()) {
+      sb.append(circuitBreakerType.toString() + " " + circuitBreakerMap.get(circuitBreakerType).printDebugInfo());
+    }
+
+    return sb.toString();
+  }
+
+  /**
+   * Register default circuit breakers and return a constructed CircuitBreakerManager
+   * instance which serves the given circuit breakers.
+   *
+   * Any default circuit breakers should be registered here
+   */
+  public static CircuitBreakerManager buildDefaultCircuitBreakerManager(SolrCore solrCore) {
```

Comment:
Do we actually need `SolrCore` here, or just the breakers' config (currently in `SolrConfig`)?